dhis-2/dhis-web-api/src/main/resources/openapi/EventsExportController.md

```diff
@@ -131,3 +131,71 @@ Supported properties are `assignedUser`, `assignedUserDisplayName`, `attributeOp
 `completedBy`, `createdAt`, `createdBy`, `deleted`, `enrolledAt`, `enrollment`, `enrollmentStatus`, `event`, `followup`,
 `occurredAt`, `orgUnit`, `orgUnitName`, `program`, `programStage`, `scheduleAt`, `status`, `storedBy`, `trackedEntity`,
 `updatedAt`, `updatedBy`.
+
+### `*.parameter.EventRequestParams.fields`
+
+Get only the specified fields in the JSON response. This query parameter allows you to remove unnecessary fields from
+the JSON response and in some cases decrease the response time. Refer to
+https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-master/metadata.html#webapi_metadata_field_filter
+for how to use it.
+
+NOTE: this query parameter has no effect on a CSV response!
+
+### `*.parameter.EventRequestParams.filter`
+
+`<filter1>[,<filter2>...]`
+
+Get events matching given filters on data values. A filter is a colon separated data element UID with optional
+operator and value pairs. Example: `filter=H9IlTX2X6SL:sw:A` with operator starts with `sw` followed by a value. Special
```

Comment:
What does it mean `optional` operator and value pairs?
Can we just filter on a data value? If yes, what is the expected result?

Reply:
its optional but I don't know what is expected. its weird is what I see 😂 maybe someone else knows? at least when I tried it did not seem to mean, return events that have a data value with this data element.

Reply:
I didn't know that the operator and value pairs was optiona. It sounds sensible to allow this as a syntax for filtering on non-null values - but is there any precedence on this in the code or other endpoints?

Reply:
@enricocolasante @Bekkalizer the operator is optional but it does not work in a way that makes sense to me.

Here I get events with a data value of data element `DX4LVYeP7bw` equal to 50

```
curl --silent -u system:System123 -H 'content-type: application/json' 'https://play.dhis2.org/dev/api/tracker/events?orgUnit=DiszpKrYNg8&page=1&pageSize=1&fields=event,dataValues&filter=DX4LVYeP7bw:EQ:50' | jq
{
  "page": 1,
  "pageSize": 1,
  "instances": [
    {
      "event": "SObENdEf76z",
      "dataValues": [
        {
          "createdAt": "2019-08-21T13:27:26.629",
          "updatedAt": "2019-09-27T00:01:48.287",
          "storedBy": "system",
          "providedElsewhere": false,
          "dataElement": "DX4LVYeP7bw",
          "value": "50"
        },
        {
          "createdAt": "2019-08-21T13:27:26.629",
          "updatedAt": "2019-09-27T00:01:48.287",
          "storedBy": "system",
          "providedElsewhere": false,
          "dataElement": "k0rev4WSffi",
          "value": "20"
        }
      ]
    }
  ]
}
```

the same request without operator

```
curl --silent -u system:System123 -H 'content-type: application/json' 'https://play.dhis2.org/dev/api/tracker/events?orgUnit=DiszpKrYNg8&page=1&pageSize=1&fields=event,dataValues&filter=DX4LVYeP7bw' | jq    {
  "page": 1,
  "pageSize": 1,
  "instances": [
    {
      "event": "rBjxtO8npTb",
      "dataValues": []
    }
  ]
}
```

Reply:
So if we think optional operators could have a use we would need to define that first and implement it accordingly. Should we thus hide the fact that the operator is optional from the docs for now? Then figure out how it should behave, implement and document it.

Reply:
Agreed - I think that is a good approach.